### PR TITLE
Eight keys resource completed.

### PR DESCRIPTION
**Pull request by mphprogrammer**
Same general changes as the Va Crosswalk resouce

### CODE REVIEW

**Comment by mphprogrammer** [comment]
I missed some specs, now added. Also - corrected a mistake in va-crosswalk and weams controller spec, as well as a delete link for va-controller.

**Comment by mphprogrammer** [comment]
Again Jeff - let me know if this is too big. It is similar in all respects (nearly identical in form save one line in the populate method of The EightKey model) as the VaCrosswalk and Weam resources.

**Comment by mphprogrammer** [comment]
Let me know if you rather I break up the PRs action by action.

File: Gemfile [hunk 1]
```diff
@@ -47,6 +47,9 @@ gem 'puma', '~> 2.16'
 # Used to colorize output for rake tasks
 gem "rainbow"
 
+# Carmen for state names
+gem 'carmen-rails', '~> 1.0', '>= 1.0.1'
```
Comment (plusjeff) [inline Gemfile]: `carmen-rails` is no longer maintained. Are there alternatives?

https://github.com/jim/carmen-rails

File: app/assets/javascripts/eight_keys.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
```
Comment (plusjeff) [inline app/assets/javascripts/eight_keys.js]: Can these auto-gen files get removed?

File: app/assets/javascripts/va_crosswalks.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
```
Comment (plusjeff) [inline app/assets/javascripts/va_crosswalks.js]: Can these auto-gen files get removed?

File: app/assets/stylesheets/eight_keys.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the EightKeys controller here.
```
Comment (plusjeff) [inline app/assets/stylesheets/eight_keys.scss]: Can these auto-gen files get removed?

File: Gemfile (hunk 1 above)
Comment (mphprogrammer) [inline Gemfile]: Off hand do you know of a gem for a list of states?

File: app/assets/javascripts/eight_keys.js (hunk 2 above)
Comment (mphprogrammer) [inline app/assets/javascripts/eight_keys.js]: Removed them.

File: app/assets/stylesheets/eight_keys.scss (hunk 4 above)
Comment (mphprogrammer) [inline app/assets/stylesheets/eight_keys.scss]: Removed.

File: app/helpers/eight_keys_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module EightKeysHelper
```
Comment (plusjeff) [inline app/helpers/eight_keys_helper.rb]: Can get removed

Comment (mphprogrammer) [inline app/helpers/eight_keys_helper.rb]: Removed

**Comment by jeffmaher** [comment]
Is the first line of the eight keys test data valid?
https://github.com/department-of-veterans-affairs/gibct-data-service/blob/eight-keys/spec/test_data/eight_key_test.csv

**Comment by mphprogrammer** [comment]
Unfortunately, that is the way the eight-keys file comes from the VA/Edu - Patrick has tried to change it, but to no avail. I would rather that the stakeholders not have to massage csvs if it can be helped.

**Comment by jeffmaher** [comment]
It test fine to me. LGTM.

**Pull request merged**